apps/vue/src/challenges/stack/index.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+    <div class="app">
+        <div>
+            <input v-model="userInput" placeholder="Enter the value" style=" width: 100%;" />
+            <div class="action-buttons">
+                <button @click="pushItem">Push</button>
+                <button @click="popItem">Pop</button>
+                <button @click="peekItem">Peek</button>
+                <button @click="isStackEmpty">IsEmpty</button>
+                <button @click="isStackFull">IsFull</button>
+            </div>
+        </div>
+        <hr />
+        <h4 class="text-center">{{ output }}</h4>
+        <ul class="text-center">
+            <li v-for="(value, index) in stack" :key="index">
+                {{ value }}
+            </li>
+        </ul>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { ref, Ref } from 'vue'
+
+const stack: Ref<string[]> = ref([])
+const userInput: Ref<string> = ref('')
+const output: Ref<string> = ref('')
+
+function pushItem() {
+    if (stack.value.length == 5) {
+        output.value = 'Stack is full'
+    } else if (userInput.value) {
+        stack.value.push(userInput.value)
+        output.value = `${userInput.value} is pushed into the Stack`
+        userInput.value = ''
+    } else {
+        output.value = 'Please enter value first'
+    }
+}
+
+function popItem() {
+    if (!stack.value.length) {
+        output.value = 'Stack is empty'
+    } else {
+        output.value = `${stack.value[stack.value.length - 1]} is popped from the Stack`
+        stack.value = stack.value.slice(0, -1)
```

Comment:
Will the `.pop()` method work instead of `.slice(0, -1)`?

Reply:
Updated this change